src/main/resources/mappers/clickhouse/sample/ClickhouseSampleMapper.xml

```diff
@@ -38,4 +98,198 @@
+            <property name="projection" value="DETAILED"/>
+        </include>
+    </select>
+
+    <select id="getSamplesBySampleListIds" resultMap="SampleResultMap">
+        <include refid="selectSample" />
+        <include refid="filterBySampleListId" />
+    </select>
+
+    <select id="getSummarySamplesBySampleListIds" resultMap="SummarySampleResultMap">
+        <include refid="selectSummarySample" />
+        <include refid="filterBySampleListId" />
+    </select>
+
+    <select id="getDetailedSamplesBySampleListIds" resultMap="DetailedSampleResultMap">
+        <include refid="selectDetailedSample" />
+        <include refid="filterBySampleListId" />
+    </select>
+
+    <select id="getMetaSamples" resultType="org.cbioportal.legacy.model.meta.BaseMeta">
+        <include refid="selectMetaSample" />
+        <include refid="filterByCaseAndStudy"/>
+    </select>
+
+    <select id="getMetaSamplesBySampleListIds" resultType="org.cbioportal.legacy.model.meta.BaseMeta">
+        <include refid="selectMetaSample" />
+        <include refid="filterBySampleListId" />
+    </select>
+
+    <select id="getSample" resultMap="DetailedSampleResultMap">
+        <include refid="selectDetailedSample" />
+        <include refid="filterBySampleAndStudyId" />
+    </select>
+    
+    <sql id="sampleColumnsId">
+        sample_derived.internal_id as internalId,
+        sample_stable_id as stableId,
+        patient_stable_id as patientStableId,
+        sample_derived.cancer_study_identifier as cancerStudyIdentifier,
+        sample_unique_id_base64 as uniqueSampleKey,
+        patient_unique_id_base64 as uniquePatientKey
+    </sql>
+    
+    <sql id="sampleColumnsSummary">
+        <include refid="sampleColumnsId" />,
+        sample_type as sampleType,
+        patient_internal_id as patientId
+    </sql>
+
+    <sql id="sampleColumnsDetailed">
+        <include refid="sampleColumnsSummary" />,
+        sequenced as sequenced,
+        copy_number_segment_present as copyNumberSegmentPresent,
+        patient_internal_id as "patient.internalId",
+        patient_stable_id as "patient.stableId",
+        cs.cancer_study_id as "patient.cancerStudyId",
+        cs.cancer_study_identifier as "patient.cancerStudyIdentifier",
+        cs.cancer_study_id as "patient.cancerStudy.cancerStudyId",
+        cs.cancer_study_identifier as "patient.cancerStudy.cancerStudyIdentifier",
+        cs.type_of_cancer_id as "patient.cancerStudy.typeOfCancerId",
+        cs.name as "patient.cancerStudy.name",
+        cs.description as "patient.cancerStudy.description",
+        cs.public as "patient.cancerStudy.publicStudy",
+        cs.pmid as "patient.cancerStudy.pmid",
+        cs.citation as "patient.cancerStudy.citation",
+        cs.groups as "patient.cancerStudy.groups",
+        cs.status as "patient.cancerStudy.status",
+        cs.import_date as "patient.cancerStudy.importDate",
+        rg.name as "patient.cancerStudy.referenceGenome"
+    </sql>
```

Comment:
So, I would say this is a good abstraction but the extra abstraction makes it easier to get lost.

The balance is to not duplicate code, but also don't abstract so much